### PR TITLE
Add docs for run dapr without cli.

### PR DESCRIPTION
**Pull request by abogdanov37**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Contains documentation which describe how to run Dapr in Kubernetes without CLI or Helm and without operator.

## Issue reference

dapr/dapr#2865

### CODE REVIEW

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy-without-cli.md
Comment (AaronCrawfis) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy-without-cli.md]:
```suggestion
type: docs
title: "Deploy Dapr on a Kubernetes cluster without CLI"
linkTitle: "Deploy Dapr without CLI"
weight: 21000
description: "Follow these steps to deploy Dapr at Kubernetes without cluster permissions and the CLI"
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy-without-cli.md
Comment (AaronCrawfis) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy-without-cli.md]:
```suggestion
This page describes the process of installation Dapr without CLI or Helm when you have no cluster permissions. Such situation may arise when you have a project in OKD or a namespace in Kubernetes. In this case you have no cluster permission and can't deploy the operator. You can't also address cluster admin to install Dapr operator because you are the only one who needs it.
```

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]

**Comment by AaronCrawfis** [comment]
@yaron2 could you take a look at this please?

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy-without-cli.md
Comment (msfussell) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy-without-cli.md]: Typo Kubernes should be Kubernetes. 

I suggest this sentence is reworded to  "Dapr is installed as a Kubernetes deployment."

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy-without-cli.md
Comment (msfussell) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy-without-cli.md]: Suggest reword this to
"To run the application use the **dns** mode . In this mode to locate the Dapr application, use services with suffix `-dapr`.

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy-without-cli.md
Comment (msfussell) [reply daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy-without-cli.md]: Just to be clear, this example only deploys a local development environment to the kubernetes cluster, so this should be  explicitly mentioned in the title. This is not about setting up a production environment which should use Helm or the CLI. This is not clear in the overview and it would be good to add.

Comment (abogdanov37) [reply daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy-without-cli.md]: I agree with you about example. But you could deploy three instances of placement and dashboard. It will be similar to production environment

**Comment by orizohar** [comment]
@msfussell @AaronCrawfis @yaron2 - Looks like comment are addressed, can you approve or provide more comments?

**Review by msfussell — COMMENTED** [review]
When I read this article, I am unsure of the scenario, whether this is targeting just development scenarios or for production? The article needs to says when to do this, as I presume that this is *only* for local development, given that only the Dapr Placement service is deployed and not the other three Dapr control  plane services. Can you explain more your intention of when someone would use this?

**Comment by yaron2** [comment]
> When I read this article, I am unsure of the scenario, whether this is targeting just development scenarios or for production? The article needs to says when to do this, as I presume that this is _only_ for local development, given that only the Dapr Placement service is deployed and not the other three Dapr control plane services. Can you explain more your intention of when someone would use this?

The assumption that you need the control plane components to run in production is false.

In self hosted mode, you can run production workloads with only the Dapr sidecar process and nothing else, depending on your use case you can add optional components (placement if you need actors, sentry if you need mTLS).

**Comment by abogdanov37** [comment]
> When I read this article, I am unsure of the scenario, whether this is targeting just development scenarios or for production? The article needs to says when to do this, as I presume that this is _only_ for local development, given that only the Dapr Placement service is deployed and not the other three Dapr control plane services. Can you explain more your intention of when someone would use this?

**The problem** is that we faced a situation when we need to run DAPR in the OpenShift project without the ability to get all permissions to run the operator. 
**The solution** is a new mode for Dapr runtime. It loads configuration from the file as Standalone mode and uses DNS instead of mDNS to resolve app location as in Kubernetes mode. 

I think this solution can be used in production because we can many instances of placement, sentry and other) In the article, I gave an example of how to use mode.

**Comment by abogdanov37** [comment]
Could you reopen this PR or I should create a new one?

**Comment by AaronCrawfis** [comment]
> Could you reopen this PR or I should create a new one?

I'll reopen & make sure to review with the maintainers tomorrow during our triage meeting

**Comment by abogdanov37** [comment]
> > Could you reopen this PR or I should create a new one?
> 
> I'll reopen & make sure to review with the maintainers tomorrow during our triage meeting

Thanks a lot. I address all comments.

**Comment by AaronCrawfis** [comment]
Converting to draft until https://github.com/dapr/dapr/pull/2903 is resolved

**Comment by abogdanov37** [comment]
Guys, I address all comments. Have any concerns about this PR?

**Comment by AaronCrawfis** [comment]
> Guys, I address all comments. Have any concerns about this PR?

@abogdanov37 we're still waiting on https://github.com/dapr/dapr/pull/2903 to merge before we merge the docs, which is waiting on https://github.com/dapr/components-contrib/pull/743. Once those two are merged we can merge this. Thanks!

**Comment by AaronCrawfis** [comment]
Still waiting on https://github.com/dapr/dapr/pull/2903

**Comment by yaron2** [comment]
@abogdanov37 I just want to get crisp on the implementation here. 

You allow Dapr to internally discover the k8s services of other Dapr instances with this DNS pattern: ```myid-dapr.abc.svc:1234```.

Is this not identical to the regular Kubernetes DNS? Dapr today allows you to find and discover services both in the namespace of the caller app and also in a different namespace.

**Comment by abogdanov37** [comment]
> @abogdanov37 I just want to get crisp on the implementation here.
> 
> You allow Dapr to internally discover the k8s services of other Dapr instances with this DNS pattern: `myid-dapr.abc.svc:1234`.
> 
> Is this not identical to the regular Kubernetes DNS? Dapr today allows you to find and discover services both in the namespace of the caller app and also in a different namespace.
Hi! I wrote about it earlier. 
1. In Kubernetes name resolver added suffix `cluster.local` which is not common
2. I need the ability to run solution without Kubernetes and without mDNS for this I add PR https://github.com/dapr/components-contrib/pull/858

**Comment by AaronCrawfis** [comment]
Closing as dapr/dapr issue was closed

**Pull request closed without merging**
